web/src/components/core/FileViewer.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from "react";
+import { Popup } from "~/components/core";
+import { Alert, Bullseye, Spinner } from "@patternfly/react-core";
+import cockpit from "../../lib/cockpit";
+
+import "./fileviewer.scss";
+
+// file loading indicator
+const spinner = () => {
+  return (
+    <Bullseye>
+      <Spinner size="xl" />
+    </Bullseye>
+  );
+};
+
+export default function FileViewer({ file, title, onCloseCallback }) {
+  // the popup is visible
+  const [isOpen, setIsOpen] = useState(true);
+  // error message for failed load
+  const [error, setError] = useState(null);
+  // the file content
+  const [content, setContent] = useState("");
+  // current state
+  const [state, setState] = useState("init");
+
+  useEffect(() => {
+    if (state !== "init") return;
```

Comment:
I have to test it, but I think you do not need that _"init" state_. If you do not use dependencies in the `useEffect` it will be executed only once. So, you drop the _state_ thingy and directly relay in _error_ and _content_. That is, displaying the spinner until either, error or content is not null.